ds3-sdk/src/main/java/com/spectralogic/ds3client/commands/parsers/utils/ResponseParserUtils.java

```diff
@@ -145,4 +149,87 @@ private static String readResponseString(final WebResponse response) throws IOEx
             return writer.toString();
         }
     }
+
+    /**
+     * Retrieves the blob checksum type from the response headers. If there is no blob checksum
+     * type header, than NONE is returned. If there is more than one value for the blob checksum
+     * type header, than an illegal argument exception is thrown.
+     */
+    public static ChecksumType.Type getBlobChecksumType(final Headers headers) {
+        final List<String> types = headers.get(BLOB_CHECKSUM_TYPE_HEADER);
+        switch (types.size()) {
+            case 0:
+                return ChecksumType.Type.NONE;
+            case 1:
+                return toBlobChecksumType(types.get(0));
+            default:
+                throw new IllegalArgumentException(String.format("Response has more than one header value for '%s'", BLOB_CHECKSUM_TYPE_HEADER));
+        }
+    }
+
+    /**
+     * Converts the response header value for {@code BLOB_CHECKSUM_TYPE_HEADER} into a {@code ChecksumType.Type}
+     */
+    static ChecksumType.Type toBlobChecksumType(final String checksumType) {
+        if (Guard.isStringNullOrEmpty(checksumType)) {
+            return ChecksumType.Type.NONE;
+        }
+        switch (checksumType) {
+            case "CRC_32":
+                return ChecksumType.Type.CRC_32;
```

Comment:
You should be able to use the built in `ChecksumType.Type.valueOf(checksumType)` method to do this same logic.